### ISSUE TITLE
Dynamic elements

### ISSUE THREAD

**Issue by bhack** [facebookresearch/vggt#13]
Same issue from our topic at vggsfm https://github.com/facebookresearch/vggsfm/issues/9#issuecomment-2251009129

**Comment by jytime** [comment]
Thanks for your interest in our series of work! The simplest solution here is the same to what we have provided for vggsfm, using a mask to filter out the dynamic pixels, as linked below

https://github.com/facebookresearch/vggsfm/tree/main?tab=readme-ov-file#8-dynamicmoving-objects

You can simply masking the dynamic pixels to 0/1 and the model will ignore them. 

Besides, we found that VGGT can be easily finetuned to also reconstruct dynamic pixels, similar to what have been done in MonST3R and CUT3R. We will share the training script so people can try to tune on their own kind of data.

**Comment by bhack** [comment]
> Besides, we found that VGGT can be easily finetuned to also reconstruct dynamic pixels, similar to what have been done in MonST3R and CUT3R.

Yes in this specific case/repo I was talking about this

**Comment by WenjiaWang0312** [comment]
Dear author @jytime , thank you for your help.  Sorry for opening this issue again. But I am wondering why the model can ignore the pure 0 pixels. Do you mean the trained model naturally can view pure 0 pixels as non-sense holes? Or this is based on any post-processing rules?

**Comment by lcc-dr** [comment]
Hi @jytime, if I want to get the depth of dynamic pixel and I only have sparse depth ground truth(e.g. from lidar projection), can vggt also be easily finetuned?

thanks a lot for your help.

**Comment by jytime** [comment]
Hi @lcc-dr it should work as our training datasets include waymo, whose gt is also sparse (from lidar). But this may need to freeze some layers I guess

**Comment by rocket-ycyer** [comment]
Why is this not working? @jytime 
![Image](https://github.com/user-attachments/assets/dadab7c8-e7f1-499d-b39a-108f7c57530f)

**Comment by lucacarlone** [comment]
> Thanks for your interest in our series of work! The simplest solution here is the same to what we have provided for vggsfm, using a mask to filter out the dynamic pixels, as linked below
> 
> https://github.com/facebookresearch/vggsfm/tree/main?tab=readme-ov-file#8-dynamicmoving-objects
> 
> You can simply masking the dynamic pixels to 0/1 and the model will ignore them.
> 
> Besides, we found that VGGT can be easily finetuned to also reconstruct dynamic pixels, similar to what have been done in MonST3R and CUT3R. We will share the training script so people can try to tune on their own kind of data.

Hi @jytime - thanks for sharing this! did you end up releasing scripts to finetune in dynamic scenes?
